src/resources/cpp/language-basics.md

```diff
@@ -1,15 +1,307 @@
+the various floating point types.
+
+::: info IEEE-754 Floating Point Model
+
+Because C++ needs to support many different types of computers, it does not enforce that floating points are represented
+in a specific way. For the sake of this tutorial, we will assume that you are working on a device that uses the IEEE-754
+floating point specification. This is the case for almost all consumer hardware. The rest of the tutorial will make the
+assumption that a `float` is an IEEE-754 binary32 format float and that a `double` is an IEEE-754 binary64 format float.
+
+| Type     | Range                  | Width |
+| -------- | ---------------------- | ----- |
+| `float`  | $\pm 3.402 * 10^{38}$  | 32    |
+| `double` | $\pm 1.797 * 10^{308}$ | 64    |
+
+The type `long double` has the restriction that its range must be at least as large as a `double`. On many x86-64
+platforms, it is implemented as an IEEE-754 binary64 extended format, where the width of the type is 80 bits. However,
+MSVC is an exception to this rule, where it is only 64 bits. On other platforms, it may be implemented as a IEEE-754
+binary128 format float.
+
+:::
+
+### Literals
+
+Literals are a way that C++ allows us to provide a value directly in the code. You may recall above where the code
+snippets were creating variables using numbers in the code. Those numbers are referred to as literals.
+
+Integer literals are just numbers in code with no trailing decimal portion. By adding a `u`, you can tell the compiler
+that the literal represents an unsigned integer type. Floating point literals are numbers in code with a trailing
+decimal portion. Without a `f` at the end, the compiler will treat the number as a `double`. By adding a trailing `f`,
+the compiler now treats it as a `float.
+
+```cpp
+int foo = 3; // This is an integer literal
+int ufoo = 3u; // This is an unsigned integer literal
+float bar = 3.14f; // This is a floating point literal for a float
+double baz = 3.14; // This is a floating point literal for a double
+```
+
 ## Basic Operators
+
+Operators are a type of expression that uses symbols to provide meaning. For example, `+` is an operator. As you may
+have assumed, `+` is typically used to add values together. Here, we'll introduce a handful of operators that you can
+use with the types you learned about above:
+
+| Operator | Name           | Function                                  |
+| -------- | -------------- | ----------------------------------------- |
+| `+`      | Addition       | Adds two values                           |
+| `-`      | Subtraction    | Subtracts two values                      |
+| `*`      | Multiplication | Multiplies two values                     |
+| `/`      | Division       | Divides two values                        |
+| `%`      | Remainder      | Returns the remainder of integer division |
+| `=`      | Assignment     | Assigns a value to a variable             |
```

Comment:
The C++ standard doesn't really define a "Name" for these, and saying that "Addition" "Adds two values" is pretty redundant. It would suffice to have a single column called "Meaning" that explains what it does.

You also have the problem that `+` and `-` are unary operators as well, so they don't necessarily add/subtract *two* values.